// Copyright 2017 The TensorFlow Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto2";

package toco;

// IODataType describes the numeric data types of input and output arrays
// of a model.
enum IODataType {
  IO_DATA_TYPE_UNKNOWN = 0;

  // Float32, not quantized
  FLOAT = 1;

  // Uint8, quantized
  QUANTIZED_UINT8 = 2;

  // Int32, not quantized
  INT32 = 3;

  // Int64, not quantized
  INT64 = 4;

  // String, not quantized
  STRING = 5;

  // Int16, quantized
  QUANTIZED_INT16 = 6;

  // Boolean
  BOOL = 7;
}
